File: Unorder of Operations/Unorder of Operations.rs
fn evaluate(numbers: &[i32], operators: &[char]) -> i32 {
    if numbers.is_empty() {
        return 0;
    }

    let mut result = numbers[0];

    for (i, &num) in numbers.iter().enumerate().skip(1) {
        let op = operators[(i - 1) % operators.len()];

        match op {
            '+' => result += num,
            '-' => result -= num,
            '*' => result *= num,
            '/' => result /= num, // integer division
            '%' => result %= num,
            _ => (), // ignore unknown operators
        }
    }

    result
}

fn main() {
    let numbers = vec![10, 5, 2];
    let operators = vec!['+', '*'];
    let result = evaluate(&numbers, &operators);
    println!("Result: {}", result); // Output: 30
}

File: Unorder of Operations.js
function evaluate(numbers, operators) {
    if (numbers.length === 0) return 0;

    let result = numbers[0];

    for (let i = 1; i < numbers.length; i++) {
        const op = operators[(i - 1) % operators.length];
        const num = numbers[i];

        switch (op) {
            case '+':
                result += num;
                break;
            case '-':
                result -= num;
                break;
            case '*':
                result *= num;
                break;
            case '/':
                result = Math.floor(result / num); // integer division
                break;
            case '%':
                result %= num;
                break;
        }
    }

    return result;
}

